### PR TITLE
fix(cli-repl): Detect containerized environment and disable telemetry MONGOSH-1278

### PR DESCRIPTION
**Pull request by gribnoysup**
I'm still adding some tests, but decided to open this PR already as I wasn't sure if that's the best place to put this logic and super open to feedback on this

### CODE REVIEW

File: packages/cli-repl/src/cli-repl.ts [hunk 1]
```diff
@@ -155,6 +156,29 @@ class CliRepl implements MongoshIOProvider {
...
+  async getIsContainerizedEnvironment() {
+    // Check for dockerenv file first
+    try {
+      await fs.stat('/.dockerenv');
+      return true;
+    } catch {
+      try {
+        // Check if there is any mention of docker / lxc / k8s in control groups
+        const cgroup = await fs.readFile('/proc/self/cgroup', 'utf8');
+        return /docker|lxc|kubepods/.test(cgroup);
```
Comment (addaleax) [inline packages/cli-repl/src/cli-repl.ts:168]:
```suggestion
        return /\b(docker|lxc|kubepods)\b/.test(cgroup);
```

File: packages/cli-repl/src/cli-repl.ts [hunk 2]
```diff
@@ -155,6 +156,29 @@ class CliRepl implements MongoshIOProvider {
...
+      } catch {
+        return false;
+      }
+    }
+  }
+
+  get forceDisableTelemetry() {
```
Comment (addaleax) [inline packages/cli-repl/src/cli-repl.ts:175]:
```suggestion
  get forceDisableTelemetry(): boolean {
```

File: packages/cli-repl/src/mongosh-repl.ts [hunk 3]
```diff
@@ -167,6 +167,10 @@ class MongoshNodeRepl implements EvaluationListener {
     this.runtimeState().instanceState.isInteractive = value;
   }
 
+  get isInteractive() {
```
Comment (addaleax) [inline packages/cli-repl/src/mongosh-repl.ts:170]:
```suggestion
  get isInteractive(): boolean {
```

**Review by addaleax — APPROVED** [review]
Looks good!

**Comment by gribnoysup** [comment]
Not sure what's up with the failing tests. Seems like two machines can't download required server version? Will look into this more

**Pull request merged**